graphistry/feature_utils.py

```diff
@@ -904,7 +904,14 @@ def process_dirty_dataframes(
 
         logger.info(":: Encoding DataFrame might take a few minutes ------")
         
-        X_enc = data_encoder.fit_transform(ndf, y)
+        try:
+            X_enc = data_encoder.fit_transform(ndf, y)
+        except TypeError:
+            nndf = ndf.copy()
+            object_columns = nndf.select_dtypes(include=['object']).columns
+            nndf[object_columns] = nndf[object_columns].astype(str)
+            X_enc = data_encoder.fit_transform(nndf, y)
+            logger.info("obj columns: %s are being converted to str", object_columns)
```

Comment:
can we get a test that triggers & handles this edge case?

can land after, looks good!